lib/link_checker/uri_checker/http_checker.rb

```diff
@@ -284,10 +284,20 @@ def gov_uk_uri?
       Plek.new.website_root.include?(uri.host)
     end
 
-    def additional_connection_headers
-      return nil unless gov_uk_uri?
+    def rate_limit_header
+      return {} unless gov_uk_uri?
+      { "Rate-Limit-Token": Rails.application.secrets.rate_limit_token }
+    end
 
-      { 'Rate-Limit-Token': Rails.application.secrets.rate_limit_token }
+    def basic_authorization_header
+      return {} unless LinkCheckerApi.authed_domains.include?(uri.host)
+      { "Authorization": LinkCheckerApi.authed_domains[uri.host] }
```

Comment:
I think we should rename `authed_domains` to `basic_authorization_headers` and then in this method insert the `Basic ` at the start of the string.